node/heartbeat/sender.go

```diff
@@ -104,3 +128,27 @@ func (s *Sender) SendHeartbeat() error {
 
 	return nil
 }
+
+func (s *Sender) updateMetrics(hb *Heartbeat) {
+	result := "in waiting list"
+	if s.isInEligibleList(hb.Pubkey, hb.ShardID) {
+		result = "in eligible list"
+	}
+	s.statusHandler.SetStringValue(core.MetricValidatorType, result)
+}
+
+func (s *Sender) isInEligibleList(pubkey []byte, shardID uint32) bool {
```

Comment:
this method is dupplicate with the one in monitor. Could the eligibleListProvider compute this directly?

Reply:
I would not create another structure which is only a wrapper over the nodes coordinator so I can move this method there. Another way of solving the duplicated code issue is to move this function to a common file so it can be used by both of  them, but in this case the function will have 2 additional parameters (eligible list provider and epoch handler). Which way do you prefer ?

Reply:
regarding to "What about if it is the "leaving" nodes list? Could you differentiate between eligible, waiting and leaving? And joining?":
maybe when we will use those lists also, the nodes coordinator should be able to tell us. And we will just add another method in the interface.